packages/amplify-e2e-core/src/init/amplifyPush.ts

```diff
@@ -36,6 +36,39 @@ export function amplifyPushForce(cwd: string, testingWithLatestCodebase: boolean
   });
 }
 
+/**
+ *
+ * @param cwd
+ * @param stackIdx
+ * @param testingWithLatestCodebase
+ * @returns
+ * Used to stop an iterative deployment
+ */
+export function cancelIterativeAmplifyPush(
+  cwd: string,
+  stackIdx: { current: number; max: number },
+  testingWithLatestCodebase: boolean = false,
+): Promise<void> {
+  return new Promise((resolve, reject) => {
+    spawn(getCLIPath(testingWithLatestCodebase), ['push'], { cwd, stripColors: true, noOutputTimeout: pushTimeoutMS })
+      .wait('Are you sure you want to continue?')
+      .sendLine('y')
+      .wait(`Deploying stack (${stackIdx.current} of ${stackIdx.max})`)
+      .wait(/.*UPDATE_IN_PROGRESS*/)
+      .sendLine('\x03')
+      .run((err: Error) => {
+        if (!err) {
+          resolve();
+        } else {
+          if (err.message !== `Killed the process as no output receive for 1200 Sec. The no output timeout is set to 1200`) {
+            reject(err);
+          }
+          resolve();
+        }
```

Comment:
This can be simplified a bit:

```js
if (err && err.message !== '...') {
  reject(err);
} else {
  resolve();
}
```

Reply:
agree with the simplification but that becomes a `===`, not?

Reply:
I think it will still be `!==` since we don't want to reject on that specific error message.